sources/unstructured_data/inbox/__init__.py

```diff
@@ -0,0 +1,158 @@
+                criteria.extend([f'(FROM {email_})'])
+
+        status, messages = client.uid('search', 'UID ' + str(int(last_message_num)) + ':*', *criteria)
+        message_uids = messages[0].split()
+
+        if not message_uids:
+            logger.warning("No emails found.")
+        else:
+            yield from [{"message_uid": int(message_uid)} for message_uid in message_uids]
+
+
+@dlt.transformer(name="messages", write_disposition="replace")
+def read_messages(
+    item: TDataItem,
+    host: str = dlt.secrets.value,
+    email_account: str = dlt.secrets.value,
+    password: str = dlt.secrets.value,
+) -> TDataItem:
+    message_uid = str(item["message_uid"])
+
+    with imaplib.IMAP4_SSL(host) as client:
+        client.login(email_account, password)
+        client.select()
+
+        status, data = client.uid('fetch', message_uid, '(RFC822)')
+
+        if status == 'OK':
+            raw_email = data[0]
+            if raw_email:
+                raw_email = data[0][1]
+                msg = email.message_from_bytes(raw_email)
+
+                email_data = dict(msg)
+                email_data['message_uid'] = int(message_uid)
+                email_data["date"] = pendulum.parse(msg['Date'], strict=False)
+                email_data['content_type'] = msg.get_content_type()
+                email_data['body'] = get_email_body(msg)
+                yield email_data
+
+
+@dlt.transformer(name="attachments", write_disposition="replace")
+def get_attachments_by_uid(
+    item: TDataItem,
+    storage_folder_path: str,
+    host: str = dlt.secrets.value,
+    email_account: str = dlt.secrets.value,
+    password: str = dlt.secrets.value,
+) -> TDataItem:
+    message_uid = str(item["message_uid"])
+
+    with imaplib.IMAP4_SSL(host) as client:
+        client.login(email_account, password)
+        client.select()
+
+        status, data = client.uid('fetch', message_uid, '(RFC822)')
+
+        if status == 'OK':
+            raw_email = data[0]
+            if raw_email:
+                raw_email = data[0][1]
+                msg = email.message_from_bytes(raw_email)
+
+                for part in msg.walk():
+                    content_disposition = part.get("Content-Disposition", "")
+                    if "attachment" in content_disposition:
+                        filename = part.get_filename()
+                        if filename:
+                            attachment_data = part.get_payload(decode=True)
+                            attachment_path = os.path.join(storage_folder_path, message_uid + filename)
+                            os.makedirs(os.path.dirname(attachment_path), exist_ok=True)
+
+                            with open(attachment_path, 'wb') as f:
+                                f.write(attachment_data)
+
+                            result = deepcopy(item)
+                            result.update(
+                                {
+                                    "file_name": filename,
```

Comment:
OK so this is our future standard to represent files :)
1. let's add `mtime` which is last modification time, you can take it from INTERNAL DATE
2. place the message information you get from `read_messages` in a field named `envelope`

yeah the things is getting organized